### ISSUE TITLE
Configure required googletest when with tests are off, which it shouldn't

### ISSUE THREAD

**Issue by yurivict** [InsightSoftwareConsortium/ITK#3750]
With ```-DBUILD_TESTING=OFF``` configure fails:
```
CMake Error at /usr/local/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /usr/local/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/share/cmake/Modules/FindGTest.cmake:270 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  Modules/ThirdParty/GoogleTest/itk-module-init.cmake:36 (find_package)
  CMake/ITKModuleEnablement.cmake:338 (include)
  CMakeLists.txt:542 (include)
```

Version: 5.3rc04
FreeBSD 13.1

**Comment by dzenanz** [comment]
This might be by design. Even if tests are disabled when building ITK, an externally-built (e.g. remote) module might want to enable tests for itself. For that, we need the testing infrastructure in ITK. @blowekamp @thewtex Am I right?

**Comment by blowekamp** [comment]
> This might be by design. Even if tests are disabled when building ITK, an externally-built (e.g. remote) module might want to enable tests for itself. For that, we need the testing infrastructure in ITK. @blowekamp @thewtex Am I right?

I believe the ITKTestKernel is enable for the configuration @yurivict is building, which interns pulls in the GTest dependencies. I believe your loggic is correct as to why ITKTestKernel is enabled by default.

**Comment by yurivict** [comment]
Alternatively, ITK should impose a requirement on all modules that they respect the cmake-standard [BUILD_TESTING](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Testing%20With%20CMake%20and%20CTest.html) variable.

**Comment by dzenanz** [comment]
Which module doesn't respect it?

**Comment by yurivict** [comment]
I don't know, but previous reasoning implied that some modules require GTest when ```BUILD_TESTING=OFF```.

**Comment by dzenanz** [comment]
When developing a new module (e.g. https://github.com/dzenanz/ITKIOOMEZarrNGFF), and you want to test it in different conditions (e.g. different versions of ITK, different compile options), it is advantageous to be able to enable testing only in your module, without having to compile all the other ITK's tests. That is done by turning off tests in ITK, compiling it, then configuring your module by pointing to the ITK you just compiled. That requires ITK to compile testing infrastructure (image comparison filter, GTest, image IOs etc) even if 100+ of its unit tests are not compiled.

**Comment by yurivict** [comment]
> [...] it is advantageous to be able to enable testing only in your module, without having to compile all the other ITK's tests.

I agree. You can institute an additional class of cmake variables for this, for example ```BUILD_TESTING_{module_name}```.

Modules would look for both:
```
if (BUILD_TESTING or BUILD_TESTING_MyModuleName)
  find_package(GTest)
   ...do the test...
endif()
```

Arbitrarily requiring GTest in unrelated places isn't a solution.

**Comment by dzenanz** [comment]
There isn't just one module. There are many modules, and anyone can start making a new module. So your suggested approach can't work.

**Comment by dzenanz** [comment]
What you probably want, is to disable building of the TestKernel. You should be able to accomplish that by disabling default modules (during CMake configure step), then enabling all the groups except Core, and then enabling all the modules there except `ITKTestKernel`.
